elixir/lib/contracts.ex

```diff
@@ -14,6 +14,14 @@ defmodule Trento.Contracts do
   def to_event(%mod{} = struct, opts \\ []) do
     id = Keyword.get(opts, :id, UUID.uuid4())
     source = Keyword.get(opts, :source, "trento")
+
+    time =
+      Keyword.get(
+        opts,
+        :time,
+        Google.Protobuf.Timestamp.new(seconds: System.system_time(:second))
+      )
```

Comment:
Sorry I suggested the wrong default, this should take `DateTime.utc_now()`

and then we need to transform it to seconds so

```elixir 
time = Google.Protobuf.Timestamp.new!(seconds: opts |> Keyword.get(:time) |> DateTime.to_unix())
```

Reply:
Should be good now

Reply:
I didn't take your suggestion exactly, as `Keyword.get(:time)` could actually already return an int, hence the pipe to DateTime.to_unix() would be redundant

Reply:
@rtorrero we assume that the user passes a `DateTime`. Passing an int would be not ergonomic at all.

Reply:
@rtorrero At the end, we would have this:
```
    time =
      Keyword.get(
        opts,
        :time,
        DateTime.utc_now()
      )
```

And below something like this (and this variable is the one we put in the final struct):
```
time_attr = Google.Protobuf.Timestamp.new!(seconds: time |> DateTime.to_unix())
```

Or anything like that (maybe you can put the `to_unix` pipe together with the `Keyword.get`.
At the end, the idea is that the incoming variable to be of `DateTime` type.

Reply:
Ok, should be good now

Reply:
@rtorrero  could we add a test for user options? just one test should be good and we check that `source` and `time` option are handled correctly
e.g.

"should accept options"